packages/ui/src/components/Contexts/Variables/Modals/OdbImport/useImportObservation.tsx

```diff
@@ -0,0 +1,180 @@
+import { useConfiguration, useUpdateConfiguration } from '@gql/configs/Configuration';
+import { GET_INSTRUMENT, useResetInstruments } from '@gql/configs/Instrument';
+import { useRotator, useUpdateRotator } from '@gql/configs/Rotator';
+import { useRemoveAndCreateBaseTargets, useRemoveAndCreateWfsTargets } from '@gql/configs/Target';
+import type { SourceProfile } from '@gql/odb/gen/graphql';
+import { useGetCentralWavelength, useGetGuideEnvironment } from '@gql/odb/Observation';
+
+import { extractMagnitude } from '@/Helpers/bands';
+import { firstIfOnlyOne } from '@/Helpers/functions';
+import { extractGuideTargets } from '@/Helpers/guideTargets';
+import { useToast } from '@/Helpers/toast';
+import { extractCentralWavelength } from '@/Helpers/wavelength';
+import type { ConfigurationType, OdbObservationType } from '@/types';
+
+export function useImportObservation() {
```

Comment:
With this as a separate hook, it might be good to try and add some tests. [vitest-browser-react](https://github.com/vitest-dev/vitest-browser-react) has some utilities for testing hooks